dbt/adapters/redshift/connections.py

```diff
@@ -233,27 +259,19 @@ def connect():
 class RedshiftConnectionManager(SQLConnectionManager):
     TYPE = "redshift"
 
-    def _get_backend_pid(self):
-        sql = "select pg_backend_pid()"
-        _, cursor = self.add_query(sql)
-
-        res = cursor.fetchone()
-        return res[0]
-
     def cancel(self, connection: Connection):
         try:
-            pid = self._get_backend_pid()
+            pid = connection.handle.backend_pid
```

Comment:
@holly-evans This is all looking very good but we do have some tests failing was looking at this a bit and here may be a good point doesn't look like we setup a `handle` object so we may need to ref the new ConnectionWrapper more directly.

If I can be of any more help please feel free to message me I will try to spend a little more time with this to give more advice as I can.